src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -344,133 +344,304 @@ public class SqlMutationEngine : IMutationEngine
+                        // would not be available in the URL and needs to be appened. So, the primary key of the newly created item which is stored in the primaryKeyRoute
+                        // is used to construct the Location Header.
+                        return ConstructCreatedResultResponse(mutationResultRow!.Columns, selectOperationResponse, primaryKeyRoute, isReadPermissionConfiguredForRole, isDatabasePolicyDefinedForReadAction);
+                    }
+
+                    if (context.OperationType is EntityActionOperation.Update || context.OperationType is EntityActionOperation.UpdateIncremental)
+                    {
+                        return ConstructOkMutationResponse(mutationResultRow!.Columns, selectOperationResponse, isReadPermissionConfiguredForRole, isDatabasePolicyDefinedForReadAction);
+                    }
                 }
+
             }
 
             // if we have not yet returned, record is null
             return null;
         }
 
+        /// <summary>
+        /// Constructs a FindRequestContext from the Insert/Upsert RequestContext and the results of insert/upsert database operation.
+        /// For REST POST, PUT AND PATCH API reqeusts, when there are database policies defined for the read action,
+        /// a subsequent select query that honors the database policy is executed to fetch the results.
+        /// </summary>
+        /// <param name="context">Insert/Upsert Request context for the REST POST, PUT and PATCH request</param>
+        /// <param name="mutationResultRow">Result of the insert/upsert database operation</param>
+        /// <param name="roleName">Role with which the API request is executed</param>
+        /// <returns>Returns a FindRequestContext object constructed from the existing context and create/upsert operation results.</returns>
+        private FindRequestContext ConstructFindRequestContext(RestRequestContext context, DbResultSetRow mutationResultRow, string roleName)
+        {
+            FindRequestContext findRequestContext = new(entityName: context.EntityName, dbo: context.DatabaseObject, isList: false);
+
+            // PrimaryKeyValuePairs in the context is populated using the primary key values from the
+            // results of the insert/update database operation.
+            foreach (string primarykey in context.DatabaseObject.SourceDefinition.PrimaryKey)
+            {
+                _sqlMetadataProvider.TryGetBackingColumn(context.EntityName, primarykey, out string? backingColumnName);
+                if (!string.IsNullOrEmpty(backingColumnName))
+                {
+                    findRequestContext.PrimaryKeyValuePairs.Add(backingColumnName, value: mutationResultRow.Columns[primarykey]!);
+                }
+                else
+                {
+                    findRequestContext.PrimaryKeyValuePairs.Add(primarykey, value: mutationResultRow.Columns[primarykey]!);
+                }
```

Comment:
this conditional can be simplified because `_sqlMetadataProvider.TryGetBackingColumn` returns a bool and `backingColumnName` has attribute `NotNullWhen(true)`